src/scenario.jl

```diff
@@ -938,12 +945,12 @@ function cyclone_mortality!(coral_cover, coral_params, cyclone_mortality)::Nothi
     # TODO: Move to own file.
 
     # Small class coral mortality
-    coral_deaths_small = coral_cover[coral_params.small, :] .* cyclone_mortality
-    coral_cover[coral_params.small, :] -= coral_deaths_small
+    coral_deaths_small = coral_cover[:, 1, :] .* cyclone_mortality
+    coral_cover[coral_cover, 1, :] -= coral_deaths_small
 
     # Mid class coral mortality
     coral_mid = hcat(collect(Iterators.partition(coral_params.mid, length(coral_params.small)))...)
-    for i in size(coral_mid, 1)
+    for i in 1:size(coral_mid, 1)
```

Comment:
I've fixed an issue with cyclone mortality. I also found a bug present in main. Here is a plot with RCP 4.5. 

### Branch: coral-dynamics-recruitment (No Bug fix)
<img width="599" alt="no_change_tst" src="https://github.com/open-AIMS/ADRIA.jl/assets/156630392/a12404d3-b85f-444a-b8fa-7bc4d674702e">

### Branch: coral-dynamics-recruitment (Bug fixed).
<img width="546" alt="fixed_bug" src="https://github.com/open-AIMS/ADRIA.jl/assets/156630392/a0d770d8-be96-4b27-a70c-55158ce6b62c">

### Branch: multidimensional-cover (Bug fix)
![image](https://github.com/open-AIMS/ADRIA.jl/assets/156630392/c8a373e6-a45e-4a74-b2ee-bfd0cfb5f669)